ros_buildfarm/config/release_build_file.py

```diff
@@ -100,9 +101,16 @@ def __init__(self, name, data):  # noqa: D107
         assert 'upload_credential_id' in data
         self.upload_credential_id = data['upload_credential_id']
 
-    def filter_packages(self, package_names):
+    def filter_packages(self, package_names, remove_implicit_distro=None):
+        # TODO(tfoote) get the distro from the constructor instead
+        # of passing it in and just have a bool flag
         res = set(package_names)
         if self.package_whitelist:
             res &= set(self.package_whitelist)
         res -= set(self.package_blacklist)
+        if remove_implicit_distro:
+            cached_pkgs = get_cached_package_manifests(remove_implicit_distro)
+            implicitly_ignored = get_implicitly_ignored_package_names(
+                cached_pkgs, self.package_blacklist)
+            res -= implicitly_ignored
```

Comment:
Until now callers of this method are doing the implicit filtering on their own. I would suggest to either do the same for the new case or instead create a function for this (outside of the build file which is only supposed to be a simple data class) and also update existing code to use that new function.

Reply:
I extended this method to support filtering on either the explicit list or what's often going to be more valuable to users the explicit + implicit list. For example we could easily leverage the implicit list on the status page. 

I can turn it into a second method that calls the first method. This is already a class with methods and not a simple data class. I can go ahead and make it into a separate method but it doesn't feel like it's significantly different from the current one, it's just a slightly different filter.

Reply:
Please move this into a separate function (in a different file) since until now this class acts as a data structure which is only concerned about the information coming from that build file and not importing any other modules. And the goal is to keep it that way to not mix other responsibilities.